Capstone/Feed/HomeViewController.m

```diff
@@ -0,0 +1,112 @@
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               NSLog(@"%@", [error localizedDescription]);
+           } else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               self.colleges = [dataDictionary objectForKey:@"colleges"];
+               for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+                   [self.collegeDictionaryArrays addObject:college];
+               }
+               [self.tableView reloadData];
+            }}];
+    [task resume];
+}
+
+-(void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (searchText.length == 0) {
+        isFiltered = false;
+    } else {
+        isFiltered = true;
+        self.filteredColleges = [[NSMutableArray alloc] init];
+        for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+            NSRange collegeName = [college[@"name"] rangeOfString:searchText options:NSCaseInsensitiveSearch];
+            if (collegeName.location != NSNotFound) {
+                [self.filteredColleges addObject:college];
+            }
+        }
+    }
+    [self.tableView reloadData];
+}
+
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if (isFiltered) {
+        return self.filteredColleges.count;
+    }
+    return self.collegeDictionaryArrays.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    HomeCell *cell = [tableView dequeueReusableCellWithIdentifier:@"HomeCell"];
+    if (isFiltered) {
+        cell.homeCollegeName.text = self.filteredColleges[indexPath.row][@"name"];
+        cell.homeCollegeDetails.text = self.filteredColleges[indexPath.row][@"shortDescription"];
+        cell.homeCollegeLocation.text = self.filteredColleges[indexPath.row][@"city"];
+        NSURL *url = [NSURL URLWithString:self.filteredColleges[indexPath.row][@"campusImage"]];
+        [cell.homeCellImage setImageWithURL:url];
+    } else {
+        cell.homeCollegeName.text = self.collegeDictionaryArrays[indexPath.row][@"name"];
+        cell.homeCollegeDetails.text = self.collegeDictionaryArrays[indexPath.row][@"shortDescription"];
+        cell.homeCollegeLocation.text = self.collegeDictionaryArrays[indexPath.row][@"city"];
+        NSURL *url = [NSURL URLWithString:self.collegeDictionaryArrays[indexPath.row][@"campusImage"]];
+        [cell.homeCellImage setImageWithURL:url];
+    }
+    return cell;
```

Comment:
There's a lot of repetitive code that can be shortened to represent the same thing.

Often, when you notice that you're typing the same thing (`self.collegeDictionaryArrays[indexPath.row]`) multiple times, it might be better to make it a variable.

Additionally, both parts of the if/else statement do the same thing: the only difference is the dictionary itself. As such, we can do something like this:

```suggestion
    HomeCell *cell = [tableView dequeueReusableCellWithIdentifier:@"HomeCell"];
    NSDictionary<NSString *, NSString *> *college;
    if (isFiltered) {
       college = self.filteredColleges[indexPath.row];
    } else {
       college = self.collegeDictionaryArrays[indexPath.row];
    }
    cell.homeCollegeName.text = college[@"name"];
    cell.homeCollegeDetails.text = college[@"shortDescription"];
    cell.homeCollegeLocation.text = college[@"city"];
    NSURL *url = [NSURL URLWithString:college[@"campusImage"]];
    [cell.homeCellImage setImageWithURL:url];
    return cell;
```

Reply:
also, we could just pass in 
```
cell.college = college;
```
if the arrays contained College objects, and have the HomeCell deal with the image setting 🤔